Mathlib/FieldTheory/Subfield.lean

```diff
@@ -941,4 +910,64 @@ theorem closure_preimage_le (f : K →+* L) (s : Set L) : closure (f ⁻¹' s) 
   closure_le.2 fun _ hx => SetLike.mem_coe.2 <| mem_comap.2 <| subset_closure hx
 #align subfield.closure_preimage_le Subfield.closure_preimage_le
 
+section Commutative
+
+variable {K : Type u} [Field K] (s : Subfield K)
+
+/-- Product of a multiset of elements in a subfield is in the subfield. -/
+protected theorem multiset_prod_mem (m : Multiset K) : (∀ a ∈ m, a ∈ s) → m.prod ∈ s :=
+  multiset_prod_mem m
+#align subfield.multiset_prod_mem Subfield.multiset_prod_mem
+
+/-- Sum of a multiset of elements in a `Subfield` is in the `Subfield`. -/
+protected theorem multiset_sum_mem (m : Multiset K) : (∀ a ∈ m, a ∈ s) → m.sum ∈ s :=
+  multiset_sum_mem m
+#align subfield.multiset_sum_mem Subfield.multiset_sum_mem
+
+/-- Product of elements of a subfield indexed by a `Finset` is in the subfield. -/
+protected theorem prod_mem {ι : Type*} {t : Finset ι} {f : ι → K} (h : ∀ c ∈ t, f c ∈ s) :
+    (∏ i in t, f i) ∈ s :=
+  prod_mem h
+#align subfield.prod_mem Subfield.prod_mem
+
+instance toAlgebra : Algebra s K :=
+  RingHom.toAlgebra s.subtype
+#align subfield.to_algebra Subfield.toAlgebra
+
+/-- The `Subfield` generated by a set in a field. -/
+def commClosure (s : Set K) : Subfield K where
```

Comment:
I'm not convinced this is the right design; we don't have this for any other subject yet, right? I think I'd prefer to see it in a separate PR

Reply:
I can make this private if you want; it's just a convenient definition to prove `mem_closure_iff` in the commutative case. It turns out people have only been using `mem_closure_iff` rather than the definition of `closure` throughout mathlib, and that's why this refactor needs to fix no other files. (If it's made private then `commClosure_eq_closure` will also be private.)